Documentation/ApiOverview/Backend/JavaScript/RequireJS/Index.rst

```diff
@@ -6,27 +6,114 @@
 RequireJS in the TYPO3 Backend
 ==============================
 
-Since TYPO3 CMS 7 it is possible to use RequireJS in the backend.
-A short explanation: RequireJS enables developers to have some kind of
-dependency handling for JavaScript. The JavaScript is written as so-called
-"Asynchronous Module Definition" (AMD).
-Some libraries delivered with TYPO3 are written as modules.
+.. deprecated::  12.0
+    The RequireJS project has been discontinued_ and was therefore
+    replaced by native ECMAScript v6/v11 modules in TYPO3 v12.0. The
+    infrastructure for configuration and loading of RequireJS
+    modules is deprecated with v12.0 and will be removed in TYPO3 v13. See
+    :ref:`RequireJS to ES6 migration <requirejs-migration>`.
+
+.. _discontinued: https://github.com/requirejs/requirejs/issues/1816
 
 Credits
 =======
 
 The complete documentation about RequireJS was inspired by the
 `blog post of Andreas Fernandez <https://scripting-base.de/blog/requirejs-backend.html>`__.
 
-Overview
-========
+.. _requirejs-migration:
+
+Migration
+=========
+
+Registering modules via :php:`$pageRenderer->requireJsModules` will still
+work in TYPO3 v12. These modules will be loaded after modules registered via
+:php:`$pageRenderer->javaScriptModules`. Extensions that use
+:php:`$pageRenderer->requireJsModules` will work as before but trigger a
+PHP :php:`E_USER_DEPRECATED` error.
+
+If your extension wants to support both TYPO3 v11 and v12 you can keep the
+RequireJS version and remove it upon switching to TYPO3 v13.
+
+If you want to prevent deprecation warnings you can also implement both
+RequireJS (with a version switch) and a native ECMAScript v6/v11 (ES6) modules.
+This approach is recommended if you are working with TypeScript and the
+JavaScript will be generated anyway.
+
+..  todo: Set link to ES6 description instead of issue once
+    https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/5 is resolved
+
+Migrate your JavaScript from the AMD module format to native ES6 modules
+and register your configuration in
+:file:`EXT:my_extension/Configuration/JavaScriptModules.php`, also see
+:issue:`96510` for more information:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Configuration/JavaScriptModules.php
+
+    <?php
+
+    return [
+        'dependencies' => ['core', 'backend'],
+        'imports' => [
+            '@vendor/my-extension/' => 'EXT:my_extension/Resources/Public/JavaScript/',
+        ],
+    ];
+
+Then use :php:`TYPO3\CMS\Core\Page\PageRenderer::loadJavaScriptModules()`
+instead of :php:`TYPO3\CMS\Core\Page\PageRenderer::loadRequireJsModule()` to
+load the ES6 module:
 
-.. toctree::
-   :titlesonly:
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/SomeClass.php
 
-   Extensions/Index
-   Dependency/Index
-   Loading/Index
-   Shim/Index
+    use TYPO3\CMS\Core\Information\Typo3Version;
+    use TYPO3\CMS\Core\Page\PageRenderer;
+
+    $this->pageRenderer->loadJavaScriptModule(
```

Comment:
The `loadJavaScriptModule()` method was introduced with v12, right? So there will be a PHP error on v11, I assume?

So:
TYPO3 >= 12: use loadJavaScriptModule()
else: use loadRequireJsModule()

This way you can also have an extension compatible with v11-v13 ;-)